src/veins/modules/mobility/traci/TraCIMobility.cc

```diff
@@ -212,9 +216,13 @@ void TraCIMobility::changePosition()
             if (last_speed != -1) {
                 double acceleration = (speed - last_speed) / updateInterval;
                 double co2emission = calculateCO2emission(speed, acceleration);
+                double fuelConsumption = calculateFuelConsumption(speed, acceleration);
                 currentAccelerationVec.record(acceleration);
                 currentCO2EmissionVec.record(co2emission);
+                emit(currentCO2EmissionSignal, co2emission * updateInterval.dbl());
+                emit(currentFuelConsumptionSignal, fuelConsumption * updateInterval.dbl());
```

Comment:
By multiplying by the `updateInteval`, the emitted values are not the rate of emission/consumption anymore but rather the absolute emissions/consumption since the last update (assuming constant rate over that time). In my opinion the signal should be either renamed to reflect this behavior, or the value should be changed to output the rate instead.